File: Blockchain/Blockchain/src/main/java/no/hvl/dat159/Main.java
package no.hvl.dat159;

import java.io.IOException;
import java.io.UnsupportedEncodingException;
import java.lang.reflect.Array;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class Main {
	
	public static void main(String[] args) throws IOException {
		/*
		 * Create a blockchain and a miner, add some blocks and validate the chain.
		 * You should also System.print out the blocks as soon as the are appended,
		 * and print out the final validation result. See output.txt for example
		 * output for a solution to this assignment.
		 */
		int difficulty = 5;
		List<String> dataset = new ArrayList<String>(Arrays.asList("lol!", "lolo", "lulz", "ROFL"));
		System.out.println("DIFFICULTY: "+difficulty + "\n\n" + String.join("", Collections.nCopies(195, "#")));

		Blockchain blockchain = new Blockchain(difficulty);
		Miner m1 = new Miner(blockchain);
		for(String data : dataset)
			System.out.println("Block " + blockchain.sizeOfChain() + ": \t" + m1.createAndMineNewBlock(data));
		System.out.println(String.join("",Collections.nCopies(195, "#")));
		System.out.println("Validating blockchain...\n\n");
		try {
			Thread.sleep(2000);
		} catch (InterruptedException e) {
			e.printStackTrace();
		}
		System.out.println(blockchain.isValidChain()?"Blockchain is valid!":"Blockchain is NOT valid!");
	}


}

File: Blockchain/Blockchain/src/main/java/no/hvl/dat159/Main.kt
package no.hvl.dat159

import java.io.IOException

import java.util.ArrayList
import java.util.Arrays
import java.util.Collections

object Main {

    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val difficulty = 5
        val dataset = ArrayList(Arrays.asList("lol!", "lolo", "lulz", "ROFL"))


        println("DIFFICULTY: " + difficulty + "\n\n" + "#".repeat(195))

        val blockchain = Blockchain(difficulty)
        val m1 = Miner(blockchain)
        for (data in dataset)
            println("Block " + blockchain.sizeOfChain() + ": \t" + m1.createAndMineNewBlock(data))
        println("#".repeat(196))
        println("Validating blockchain...\n\n")
        try {
            Thread.sleep(2000)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }

        println(if (blockchain.isValidChain) "Blockchain is valid!" else "Blockchain is NOT valid!")

    }


}



